knexfile.js

```diff
@@ -1,16 +1,3 @@
-module.exports = {
-  client: 'pg',
-  connection: {
-    host: process.env.PSQL_HOST || 'localhost',
-    user: process.env.PSQL_USER || '',
-    password: process.env.PSQL_PASSWORD || '',
-    database: process.env.PSQL_DATABASE || 'openaq-local'
-  },
-  pool: {
-    min: process.env.PSQL_POOL_MIN || 2,
-    max: process.env.PSQL_POOL_MAX || 10
-  },
-  migrations: {
-    tableName: 'migrations'
-  }
-};
+require('babel-register');
+const config = require('config');
+module.exports = config.get('knex');
```

Comment:
The config module selects the file based on `NODE_ENV` variable. If you set `NODE_ENV=travis`, it will use the `travis.json` file merged over `default.json`, if the last exists.

The enviroment variables defined at `custom-environment-variables.json` take precedence over any config file. If you set `PSQL_PORT=5432` and `NODE_ENV=development` (which uses port `15432`), the database port will be set to `5432`.